lib/iex/lib/iex/evaluator.ex

```diff
@@ -254,6 +260,23 @@ defmodule IEx.Evaluator do
     :elixir_errors.parse_error(line, "iex", error, token)
   end
 
+  defp check_result(result, code) when is_list(result) do
+    if Enum.all?(result, &is_integer(&1)) do
+      input =
+        code
+        |> String.Chars.to_string
+        |> String.replace(" ", "")
+
+      string_result = String.Chars.to_string(result)
```

Comment:
We don't need to detect all cases correctly. I would do this:

1. check if the list is printable in single quotes
2. if so, check that the single quotes representation is not in the original code